### DISCUSSION TITLE
wiliwili功能请求和故障反馈笔记

### DISCUSSION THREAD

**Discussion by ghost** [xfangfang/wiliwili#344] [General]
wiliwili功能请求
使用差不多一个月了，赞。现在网页端功能太多，切换好迟钝，还是这个好用。有点杂乱，全部单开新帖太占用版面了，都写在一个帖子里好了

10:17 2024/5/2
1评论区链接/传送门不可点击没有入口
2评论区图片不可保存
3.操作模式pc违和感，没有独立的操作ui
  3.1左右键不能前进后退
  3.2单击暂停
  3.3双击窗口最大化
  3.4外壳ui按钮，全屏，最大化，参考potplayer
4网页和客户端互相交互
  4.1在网页端打开
  4.2wiliwili://协议
5DLNA接受投屏正常，发送投屏功能macast无法接收

11:18 2024/5/7
6声卡切换
7息屏播放
  7.1关闭画面
  
  
11:50 2024/5/10
评论区头像点击
up主页模式/读取投稿标签、合集列表
搜索页时光机模式，按时间段搜索
播放下一集模式切换/打开评论区或者输入文字时暂停或者停止自动下一集

11:02 2024/5/12
有点杂乱，全部单开新帖太占用版面了，都写在一个帖子里好了
C++有没有配好环境VMware镜像，挂载即用的版本，配环境好麻烦，有些插件还是utf-8的会有乱码，开启系统beta的Unicode以后其他软件又会显示异常。

12:45 2024/5/14
没有剪贴板功能，无法复制也无法粘贴
如果安装了额外的libmpv或调用异常，最好有开关
【wiliwili.exe无法找到入口，无法定位程序输入点 vkGetPhysicalDevicePorperties2于动态链接库libmpv-2.dll】
弹窗确认后可以关闭
![image](https://github.com/xfangfang/wiliwili/assets/29087287/5b997767-c819-485c-8eee-2ba73ca81fcd)

**Comment by xfangfang** [comment]
@hinum009 感谢反馈，除去新的功能请求，剩下的我可以立刻回复的有：

1. “左右键不能前进后退”：快捷键是 L R，如果要修改可以改源码，已经有不少历史的issue提过这个问题了。
2. “单击暂停”：这个应用要考虑触摸操作，不能只为鼠标用户设计（所以是双击暂停）
3. “DLNA接受投屏正常，发送投屏功能macast无法接收”：我测试是可以接收的，wiliwili内部的DLNA投屏接收写的比较简单，倒是会存在收不到的可能，其他的都应该是正常的（不过搭配上windows就可能不正常了——防火墙问题）
4. “打开评论区或者输入文字时暂停或者停止自动下一集”：现在就是这样的设置，无论设置了什么播放规则，点开评论详情时就不会自动播放下一集。
5. “C++有没有配好环境VMware镜像，挂载即用的版本，配环境好麻烦”：这段话好像不是写给wiliwili的
6. “如果安装了额外的libmpv或调用异常”：默认 mingw64 的库是不会被系统识别的，你应该是手动配置环境变量导致的问题，可以参考wiliwili的构建脚本安装支持 d3d11的libmpv，或者下载使用OpenGL版的wiliwili。

**Comment by dragonflylee** [comment]
@hinum009 windows 下的环境配置参考https://github.com/xfangfang/wiliwili/discussions/263
